### PR TITLE
Baby step towards storing type associated with type value

### PR DESCRIPTION
**Pull request by henridf**
@mccanne this is a first step toward what you outlined in #1619 and which I also need for #1785 : 

> Also, to be efficient on this front, zng.TypeType should have a zng.Type value inside of it in addition to the string.

The `eq` func in typevector/type.go should probably be lifted to `zng/type.go`. If we go this route, I would also look for any other occurences of type comparisons that may need to use this. But I wanted to see first if this route makes sense?

(And, the next step would be to fill in the Type pointer lazily for values that originated from zson, and where the zng.Type is not available at value creation time... )

### CODE REVIEW

**Comment by henridf** [comment]
(I wanted to open this as a draft  but didn't and apparently you can't switch back...)

**Comment by mccanne** [comment]
@henridf I guess the zson-parser merge is making these diffs hard to follow.  Are you proposing moving TypeOfType back into reolver.Context?  I went back and forth on that and decided it was a square peg ini a round hole and settled on this approach instead...
https://github.com/brimsec/zq/blob/master/zson/table.go
This ill let you look up any efficient lookup any zng.Value TypeValue using the zng.Type you want the TypeValue for.  So no need to have a TypeTypeValue that implements zng.Type.  e.g., see how the typoeof() function does this... https://github.com/brimsec/zq/blob/88b821dbf97f67db06e31ce797049ebec687ba1d/expr/function/function.go#L180

**Comment by henridf** [comment]
@mccanne I restored this branch to something readable after github's automatic base change... 

> This will let you look up any efficient lookup any zng.Value TypeValue using the zng.Type you want the TypeValue for. 

My goal here is to the reverse: retrieve the zng.Type given a type value zng.Value. This is to support reshape, but also things like dot-operator on type values as you outlined in #1619.

That said your comment made me realize that while I'd seen `zson.TypeTable`, I missed its `LookupType` method which means this PR shouldn't be needed. I guess `zson.(*TypeTable).LookupType()` will need a resolver.Context to create the `zng.Type` (and it should probably move to `zng/` at some point). Does this sound right?

**Comment by henridf** [comment]
>  I guess zson.(*TypeTable).LookupType() will need a resolver.Context to create the zng.Type (and it should probably move to zng/ at some point). Does this sound right?

I'll go ahead and try this on another PR...
